blazor/diagram/data-binding.md

```diff
@@ -654,105 +654,364 @@ To bind remote data to [Diagram component](https://help.syncfusion.com/cr/blazor
+        node.Width = 80;
+        node.Height = 40;
+        node.Shape = new BasicShape() { Type = Syncfusion.Blazor.Diagram.NodeShapes.Basic, Shape = NodeBasicShapes.Rectangle, CornerRadius = 8 };
+        node.Style = new ShapeStyle() { StrokeWidth = 0, Fill = "" };
+        x += 100;
+
+        Dictionary<string, object> data = node.Data as Dictionary<string, object>;
+        node.Annotations = new DiagramObjectCollection<ShapeAnnotation>() {
+            new ShapeAnnotation() {
+                Content = data["FirstName"].ToString(),
+                Style = new TextStyle() { Color = "white" }
+            }
+        };
+
+        switch (data["FirstName"].ToString()) {
+            case "Andrew":
+                node.Style.Fill = "#3A4857";
+                break;
+            case "Nancy":
+                node.Style.Fill = "#2B8C68";
+                break;
+            case "Janet":
+                node.Style.Fill = "#488CC1";
+                break;
+            case "Margaret":
+                node.Style.Fill = "#4C888F";
+                break;
+            case "Steven":
+                node.Style.Fill = "#8E4DB4";
+                break;
+            case "Laura":
+                node.Style.Fill = "#CD6A32";
+                break;
+            default:
+                node.Style.Fill = "#8E4DB4";
+                break;
         }
     }
-    private void OnConnectorCreating(IDiagramObject obj)
-    {
+
+    private void OnConnectorCreating(IDiagramObject obj) {
         Connector connector = obj as Connector;
         connector.Style.StrokeColor = "#048785";
         connector.Type = ConnectorSegmentType.Orthogonal;
         connector.TargetDecorator.Shape = DecoratorShape.None;
         connector.SourceDecorator.Shape = DecoratorShape.None;
         connector.Style = new ShapeStyle() { StrokeColor = "#3A4857", Fill = "#3A4857", StrokeWidth = 1, StrokeDashArray = "3,3" };
     }
-    private CommonElement SetTemplate(IDiagramObject node)
-    {
-        return null;
+
+}
+```
+You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/DataBinding/RemoteData)
+
+### Binding with URL services
+
+A URL service is a web service that is accessible via a URL (Uniform Resource Locator). These services provide specific functionalities or data when a client sends an HTTP request to the given URL. URL services are a foundational aspect of web-based APIs (Application Programming Interfaces) and RESTful services.The [UrlAdaptor](https://blazor.syncfusion.com/documentation/data/adaptors#url-adaptor) acts as the base adaptor for interacting with remote data services. Most of the built-in adaptors are derived from the `UrlAdaptor`.
+
+The following sample code demonstrates binding data to the Diagram component through the [SfDataManager](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Data.SfDataManager.html) using `UrlAdaptor`,
+
```

Comment:
A URL service is a web service accessible via a Uniform Resource Locator (URL). These services provide specific functionalities or data when a client sends an HTTP request to the given URL. URL services are a foundational aspect of web-based APIs (Application Programming Interfaces) and RESTful services. The [UrlAdaptor](https://blazor.syncfusion.com/documentation/data/adaptors#url-adaptor) serves as the base adaptor for interacting with remote data services. Most built-in adaptors are derived from the UrlAdaptor.

The following sample code demonstrates how to bind data to the Diagram component through the [SfDataManager](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Data.SfDataManager.html) using the UrlAdaptor: